src/Orleans/Providers/ILeaseProvider.cs

```diff
@@ -0,0 +1,94 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Orleans.LeaseProviders
+{
+    /// <summary>
+    /// Acquired lease
+    /// </summary>
+    public class AcquiredLease 
+    {
+        /// <summary>
+        /// The resource key which the lease is attached to 
+        /// </summary>
+        public string ResourceKey { get; }
+        /// <summary>
+        /// Duration of the aquired lease
+        /// </summary>
+        public TimeSpan Duration { get; }
+        /// <summary>
+        /// Lease token, whcih will be null if acquiring or renewing the lease failed
+        /// </summary>
+        public string Token { get; }
+        /// <summary>
+        /// Start time for this lease, which is when the lease is acquired or renewed
+        /// </summary>
+        public DateTime StartTimeUtc { get; }
+    }
+
+    /// <summary>
+    /// AcquireLeaseResult class, which demonstrates result of acquiring or renewing lease operation
+    /// </summary>
+    public class AcquireLeaseResult
+    {
+        /// <summary>
+        /// Acquired lease, which will be null if acquire or renew operation failed.
+        /// </summary>
+        AcquiredLease AcquiredLease { get; }
+        /// <summary>
+        /// Whether successfully acquire or renew the lease
+        /// </summary>
+        public bool Success { get; }
+        /// <summary>
+        /// If acquiring or renewing the lease failed, this is the exception which caused it. This field would be null if operation succeed.
+        /// </summary>
+        public Exception FailureException { get; }
```

Comment:
good point. With that being said, I think it may better to merge Exception and Success fields into one field, which is some sort of response state. Since exception and success are both a subset of response state. For example, RESTful API handle this situation with response code, 200 meaning succeed, 403 meaning forbidden.